server/commands.go

```diff
@@ -0,0 +1,472 @@
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	if len(data.ApprovedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no questions for this channel...",
+		}
+	}
+
+	message := "Questions:\n"
+	for index, question := range data.ApprovedQuestions[args.TeamId][args.ChannelId] {
+		message = message + fmt.Sprintf("%d.\t%s:\t%s\n", index, question.Creator, question.Question)
+	}
+
+	return &model.CommandResponse{
+		ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		Text:         message,
+	}
+}
+
+func (p *Plugin) executeCommandIcebreaker(args *model.CommandArgs) *model.CommandResponse {
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	//check if there are any approved questions yet
+	if len(data.ApprovedQuestions[args.TeamId][args.ChannelId]) == 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There are no approved questions that I can ask...",
+		}
+	}
+
+	//get a random user that is not a bot
+	user, err := p.GetRandomUser(args.ChannelId)
+	if err != nil {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "There is no user I can ask a question for...",
+		}
+	}
+
+	//build the question and ask it
+	question := data.ApprovedQuestions[args.TeamId][args.ChannelId][rand.Intn(len(data.ApprovedQuestions[args.TeamId][args.ChannelId]))]
+	message := fmt.Sprintf("Hey @%s! %s", user.GetDisplayName(""), question.Question)
+	post := &model.Post{
+		ChannelId: args.ChannelId,
+		RootId:    args.RootId,
+		UserId:    p.botID,
+		Message:   message,
+	}
+	_, err = p.API.CreatePost(post)
+	if err != nil {
+		const errorMessage = "Failed to create post"
+		p.API.LogError(errorMessage, "err", err.Error())
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         errorMessage,
+		}
+	}
+
+	return &model.CommandResponse{}
+}
+
+func (p *Plugin) executeCommandIcebreakerAdd(args *model.CommandArgs) *model.CommandResponse {
+	//check the user input and extract the question from it
+	givenQuestion := strings.TrimPrefix(args.Command, fmt.Sprintf("/%s", commandIcebreakerAdd))
+	givenQuestion = strings.TrimPrefix(givenQuestion, " ")
+	if len(givenQuestion) <= 0 {
+		return &model.CommandResponse{
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+			Text:         "Please enter a question",
+		}
+	}
+
+	newQuestion := Question{}
+	creator, _ := p.API.GetUser(args.UserId)
+	newQuestion.Creator = creator.Username
+	newQuestion.Question = givenQuestion
+
+	data := p.ReadFromStorage(args.TeamId, args.ChannelId)
+
+	//Check if the question already is proposed or even approved
+	for _, question := range data.ProposedQuestions[args.TeamId][args.ChannelId] {
+		if question.Question == newQuestion.Question {
```

Comment:
I think this comparison would check for all fields of question. I that's the case, it will report this error only when the same user adds the same question. If two users added the same question it will not detect this. Please let me know if this is not the case here.

Reply:
This compares the Question itself, not the whole struct. Should be correct as far as I can see.